### ISSUE TITLE
American Trains have hidden third color

### ISSUE THREAD

**Issue by spacek531** [OpenRCT2/OpenRCT2#5955]
The yellow parts of the American train are recolorable:
![image](https://user-images.githubusercontent.com/3454156/28405010-67377c2c-6ce0-11e7-8ebf-dcef968bd161.png)

You'd never know it, though, because there is no third color option. This train has its third color changed because this scenario doesn't have the regular trains it was built with (which have 3 colors).

**Comment by Gymnasiast** [comment]
This is something we can fix when switching to our own format.

Normally, I would say we can fix it now, but the third colour is likely saved as `0` (black), so it would break SV6 compatibility.

**Comment by spacek531** [comment]
I believe it's actually saved as `18` (yellow).

**Comment by IntelOrca** [comment]
@Gymnasiast For now you could patch the object on load like you have done with others until we have json files?

**Comment by Gymnasiast** [comment]
@IntelOrca Yes, but only if existing saved games actually have the third colour set to yellow, not 0.

**Comment by spacek531** [comment]
Here is an .sv6 with the third color in use:

[third color on train.txt](https://github.com/OpenRCT2/OpenRCT2/files/1161514/third.color.on.train.txt)

**Comment by Gymnasiast** [comment]
I changed a few bits and bobs around, but now I think I know why it wasn't enabled: there is no way to avoid the wood on the tender being redrawn as well if you opt for "all vehicles in the same colour".

It needs a bit more thought before I auto-patch the .DAT.
